apps/prairielearn/src/lib/db-types.ts

```diff
@@ -268,3 +268,34 @@ export const StripeCheckoutSessionSchema = z.object({
   subject_user_id: IdSchema.nullable(),
 });
 export type StripeCheckoutSession = z.infer<typeof StripeCheckoutSessionSchema>;
+
+export const TopicSchema = z.object({
+  color: z.string().nullable(),
+  course_id: IdSchema,
+  description: z.string().nullable(),
+  id: IdSchema,
+  name: z.string().nullable(),
+  number: z.number().nullable(),
+});
+export type Topic = z.infer<typeof TopicSchema>;
+
+export const TagsForQuestionSchema = z
```

Comment:
Rather than making this type, could we just replace the use of the `tags_for_topics()` sproc with a subquery like:
```
(SELECT to_jsonb(tags) FROM question_tags AS qt JOIN tags ON (tags.id = qt.tag_id) WHERE qt.question_id = q.id) AS tags
```
and then use a basic type like:
```
tags: Tag[],
```
Also similarly for `AssessmentsFormatForQuestionSchema` below.

We are trying to reduce the use of one-off types for sprocs (and also working towards eliminating most of our sprocs in the long term).

Reply:
There are at least a few other places that use this sproc. Long term would you rather have them all replaced by this identical subquery?

Reply:
discussed in meeting--we are going to go toward just having subqueries and getting rid of sprocs. 

I have made the changes requested.